src/lwc/utilInput/utilInput.js

```diff
@@ -149,6 +156,26 @@ export default class utilInput extends LightningElement {
         return this.type ? inputTypeByDescribeType[this.type.toLowerCase()] : TEXT;
     }
 
+    get picklistValues() {
+        if (this.picklistOptionsOverride) {
+            return this.picklistOptionsOverride;
+        }
+
+        if (this.fieldApiName === RECORD_TYPE_FIELD.fieldApiName && this.objectDescribeUtil) {
+            return this.objectDescribeUtil.getPicklistOptionsForRecordTypeIds();
+        }
+
+        return this._picklistValues;
+    }
+
+    set picklistValues(values) {
+        this._picklistValues = values;
+    }
+
+    get isRecordTypePicklist() {
+        return this.type === 'REFERENCE' && this.fieldApiName === RECORD_TYPE_FIELD.fieldApiName;
```

Comment:
Could we use a constant for `'REFERENCE'`?

Reply:
What about removing it? I think the 2nd conditional is strong enough to carry this.